### PR TITLE
[3.13] Add zizmor to pre-commit and fix most findings (#127749)

### PR DESCRIPTION
**Pull request by hugovk**
Co-authored-by: Alex Waygood <Alex.Waygood@Gmail.com>
(cherry picked from commit ae31df354d02e12bf656954c5c72380d96c1dc0e)

### CODE REVIEW

**Review by AlexWaygood — APPROVED** [review]

**Review by ezio-melotti — APPROVED** [review]

**Comment by AlexWaygood** [comment]
Looks like you might need to add `shell: bash` to a couple more windows workflows here?

**Comment by hugovk** [comment]
No, JIT/Windows is also failing in a clean 3.13 branch: https://github.com/hugovk/cpython/actions/runs/12255332851/job/34188652334
```
      | RuntimeError: Can't find clang-18!
```

And the changes in `.github/workflows/jit.yml` are only adding `persist-credentials: false` so not related to shell/variables.

cc @brandtbucher FYI

**Comment by AlexWaygood** [comment]
I think the Windows failures in #127788 might be related, however? The error message there is different

File: .github/workflows/reusable-windows-msi.yml [hunk 1]
```diff
@@ -17,8 +17,11 @@ jobs:
...
     env:
+      ARCH: ${{ inputs.arch }}
       IncludeFreethreaded: true
     steps:
     - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
     - name: Build CPython installer
-      run: .\Tools\msi\build.bat --doc -${{ inputs.arch }}
+      run: .\Tools\msi\build.bat --doc -"${ARCH}"
```
Comment (AlexWaygood) [inline .github/workflows/reusable-windows-msi.yml:27]: this probably should also have been

```suggestion
      run: .\Tools\msi\build.bat --doc -"${ARCH}"
      shell: bash
```

maybe this job doesn't run on PRs?

Comment (hugovk) [reply .github/workflows/reusable-windows-msi.yml:27]: Hmm, it ran here:

https://github.com/python/cpython/actions/runs/12255117757/job/34194444625

Comment (AlexWaygood) [reply .github/workflows/reusable-windows-msi.yml:27]: hmmm... are we absolutely confident that the environment variable was properly expanded as part of the command that successfully ran in that workflow?

Comment (hugovk) [reply .github/workflows/reusable-windows-msi.yml:27]: Let's double check. Just heading out to Helsinki Python, I can check tomorrow if no-one beats me to it!

Comment (hugovk) [reply .github/workflows/reusable-windows-msi.yml:27]: Confirmed it's needed.

[Without](https://github.com/hugovk/cpython/actions/runs/12275902522/job/34251971548):

```
Run echo 123 "${ARCH}" 456 && .\Tools\msi\build.bat --doc -"${ARCH}"
123

456
```

[With](https://github.com/hugovk/cpython/actions/runs/12275967901/job/34252185731):

```
Run echo 123 "${ARCH}" 456 && .\Tools\msi\build.bat --doc -"${ARCH}"
123 arm64 456
```

Please see PR https://github.com/python/cpython/pull/127822.

**Pull request merged**